src/Adapter/MSTest.CoreAdapter/Discovery/UnitTestDiscoverer.cs

```diff
@@ -91,11 +98,25 @@ internal void SendTestCases(string source, IEnumerable<UnitTestElement> testElem
                     navigationSessions.Add(source, PlatformServiceProvider.Instance.FileOperations.CreateNavigationSession(source));
                 }
 
+                // Get filter expression and skip discovery in case filter expression has parsing error.
+                bool filterHasError = false;
+                ITestCaseFilterExpression filterExpression = this.TestMethodFilter.GetFilterExpression(discoveryContext, logger, out filterHasError);
+                if (filterHasError)
+                {
+                    return;
```

Comment:
This would not return anything. Is this the spec?
Should we log in that case?

Reply:
I discussed with Pratap. Discovery should not happen in case filter has error